pkg/controller/account/account_controller.go

```diff
@@ -120,9 +135,34 @@ type newAwsClientInput struct {
 func (r *ReconcileAccount) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
 	reqLogger.Info("Reconciling Account")
+
+	// We expect this secret to exist in the same namespace Account CR's are created
+	awsSetupClient, err := r.getAWSClient(newAwsClientInput{
+		secretName: awsSecretName,
+		nameSpace:  awsv1alpha1.AccountCrNamespace,
+		awsRegion:  "us-east-1",
+	})
+	if err != nil {
+		reqLogger.Info("Failed to get AWS client")
+		return reconcile.Result{}, err
+	}
+
+	// before doing anything make sure we are not over the limit if we are just error
```

Comment:
Where does the error bubble up? I.e. how does any SRE team know that this condition is true?

Reply:
This goes up to the reconcile loop and errors in console.  I am putting in a better logging message that is log.Error(). Is there anything else you'd like to see

Reply:
Is there any alert via PD to someone to we know the limit has been reached?  When will SRE be paged is the question.  I don't see a metric updated here that could be used for a prometheus alert.

Reply:
Understood , We need to look over this code base again for metrics. I think that will be best suited to another PR.